hooks/caret-position.js

```diff
@@ -0,0 +1,34 @@
+
+import {useState, useRef, useEffect, useCallback} from 'react'
+
+import useFocus from './focus'
+
+function useCaretPosition({initialValue, isFocus}) {
+  const [focusedElement] = useFocus({value: null, isFocus})
+  const value = useRef(initialValue) // Save initial input value state on first render
+
+  const [start, setStart] = useState(value.current ? value.current.length : 0)
+  const [end, setEnd] = useState(value.current ? value.current.length : 0)
```

Comment:
`value` n'est pas utile, car sa valeur ne sera utilisée que pour l'initialisation de `start` et `end`. Ce qui est passé en valeur par défaut à `useState` est en effet attribué qu'une seule fois au premier cycle de vie du composant.

```suggestion
  const [start, setStart] = useState(initialValue?.current.length || 0)
  const [end, setEnd] = useState(initialValue?.current.length || 0)
```